airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1057,6 +1071,196 @@ def _delete_cluster(self, hook: DataprocHook):
+            the cluster is in desired state already.
+        """
+
+        return NotImplemented
+
+    def _get_operation(self, hook: DataprocHook) -> operation.Operation:
+        """Implement this method in child class to call the related hook method and return its result.
+
+        :param hook: Required. Instance of ``airflow.providers.google.cloud.hooks.dataproc.DataprocHook``
+            class to interact with Dataproc API
+        :return: ``google.api_core.operation.Operation`` value whether the cluster is in desired state or not
+        """
+        return NotImplemented
+
+    def _get_return_value(self, hook: DataprocHook) -> Any | None:
+        """This method can be overridden if you want to return something from the execute method.
+
+        This method's return value will be also return from the execute function.
+        This method returns None as default, if you want to return something else, then you can override it.
+        """
+
+        return None
+
+    def execute(self, context: Context):
+        hook = DataprocHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        cluster: Cluster = self._get_cluster(hook)
+        print("CLUSTER", cluster)
+        is_already_desired_state, log_str = self._check_desired_cluster_state(cluster)
+        if is_already_desired_state:
+            self.log.info(log_str)
+            return
+
+        op: operation.Operation = self._get_operation(hook)
+        hook.wait_for_operation(timeout=self.timeout, result_retry=self.retry, operation=op)
+
+        return self._get_return_value(hook)
+
+
+class DataprocStartClusterOperator(_DataprocStartStopClusterBaseOperator):
+    """Start a cluster in a project."""
+
+    operator_extra_links = (DataprocClusterLink(),)
+
+    def execute(self, context: Context):
+        self.log.info("Starting the cluster: %s", self.cluster_name)
+        super().execute(context)
+        DataprocClusterLink.persist(
+            context=context,
+            operator=self,
+            cluster_id=self.cluster_name,
+            project_id=self.project_id,
+            region=self.region,
+        )
+        self.log.info("Cluster started")
+
+    def _check_desired_cluster_state(self, cluster: Cluster) -> (bool, None | str):
+        if cluster.status.state == cluster.status.State.RUNNING:
+            return True, f'The cluster "{self.cluster_name}" already running!'
+        return False, None
+
+    def _get_operation(self, hook: DataprocHook) -> operation.Operation:
+        return hook.start_cluster(
+            region=self.region,
+            project_id=self.project_id,
+            cluster_name=self.cluster_name,
+            cluster_uuid=self.cluster_uuid,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+    def _get_return_value(self, hook: DataprocHook):
```

Comment:
Do we strongly need this function? Does the `wait_for_operation` function not return `operation.result` with cluster info?

Reply:
I tested it locally, and it seems we can just return `Cluster.to_dict(result)`  and remove this method.
At first, I was thinking that we can return Cluster information for the start operator but it is not relevant to stop operator, however returning cluster information can be useful for both cases.